### ISSUE TITLE
IllegalArgumentException: Cannot push port ZipArchiveFlow.out twice, or before it being pulled

### ISSUE THREAD

**Issue by guntiso** [akka/alpakka#2010]
```
java.lang.IllegalArgumentException: Cannot push port (ZipArchiveFlow.out(...)) twice, or before it being pulled
 	at akka.stream.stage.GraphStageLogic.push(GraphStage.scala:642)
 	at akka.stream.alpakka.file.impl.archive.ZipArchiveFlowStage$$anon$2.onUpstreamFinish(ZipArchiveFlow.scala:71)
```
Replacing `push` with `emit` in `onUpstreamFinish()` (as described in https://akka.io/blog/2016/10/21/emit-and-friends) fixes the problem.

**Comment by seglo** [comment]
Good catch @guntiso. Is this something you might be interested in creating a PR about (and creating a test for?)

**Comment by guntiso** [comment]
Ok

**Comment by ennru** [comment]
Fixed with #2090
